components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/model/graph/JsGraphBuilder.java

```diff
@@ -499,6 +502,28 @@ public String loadLocalLibrary(String functionLibraryName) throws FunctionLibrar
         return libraryScript;
     }
 
+    /**
+     * Get the secret by name.
+     *
+     * @param secretName secretName
+     * @return secretValue
+     * @throws SecretManagementException
+     */
+    public String getSecretByName(String secretName) throws SecretManagementException {
+
+        SecretResolveManager secretResolveManager = FrameworkServiceComponent.getSecretConfigManager();
+        String secretValue = null;
+
+        ResolvedSecret responseDTO = secretResolveManager.getResolvedSecret(FrameworkConstants.SECRET_TYPE, secretName);
+
+        if (responseDTO != null) {
+            secretValue = responseDTO.getResolvedSecretValue();
+        } else {
+            log.error("No secret available with " + secretName + "name.");
```

Comment:
Do we need to log this as an error?

Reply:
Here we are trying to resolve a secret which is defined in the adaptive authentication script. If that secret cannot be found, then the flow will not work as expected since the usages of that secret in the script won't work.
That's why we added this as an error log.

Reply:
Resolved with https://github.com/wso2/carbon-identity-framework/pull/5703/commits/765f3f1bcb347b0a86c0176ff61d1ef3f09e56c7